src/main/kotlin/nl/info/zac/productaanvraag/ProductaanvraagService.kt

```diff
@@ -149,33 +148,50 @@ class ProductaanvraagService @Inject constructor(
         )
 
     fun pairProductaanvraagWithZaak(productaanvraag: ModelObject, zaakUrl: URI) {
-        ZaakobjectProductaanvraag(zaakUrl, productaanvraag.url)
-            .let(zrcClientService::createZaakobject)
+        ZaakobjectProductaanvraag(zaakUrl, productaanvraag.url).let(zrcClientService::createZaakobject)
     }
 
-    fun pairAanvraagPDFWithZaak(productaanvraag: ProductaanvraagDimpact, zaakUrl: URI) {
-        val zaakInformatieobject = ZaakInformatieobject().apply {
+    fun pairAanvraagPDFWithZaak(productaanvraag: ProductaanvraagDimpact, zaakUrl: URI, throwException: Boolean = true) =
```

Comment:
Instead of suppressing exceptions I would op for using the `runCatching` and returning the `Result` directly.

That also means the other methods `createZaakInformatieobject` etc, should do the same.

Reply:
We use this function in two places - `ProductaanvraagService` and `ZaakRestService` and we want different outcomes - no exception when automated flow is running and exception when invoked by user actions.

We can instead use 2 different functions instead of the flag, but that does not change underlying implementation much.